### PR TITLE
Add hadoop-client-runtime dependency to fix quickstart NoClassDefFoundError issue

### PR DESCRIPTION
**Pull request by yashmayya**
- Currently, a number of quickstarts (`GenericQuickstart`, `MultistageEngineQuickStart` etc.) encounter errors while ingesting data into some tables:
```
java.lang.RuntimeException: Caught exception during running - org.apache.pinot.plugin.ingestion.batch.standalone.SegmentGenerationJobRunner
	at org.apache.pinot.spi.ingestion.batch.IngestionJobLauncher.kickoffIngestionJob(IngestionJobLauncher.java:152) ~[classes/:?]
	at org.apache.pinot.spi.ingestion.batch.IngestionJobLauncher.runIngestionJob(IngestionJobLauncher.java:121) ~[classes/:?]
	at org.apache.pinot.tools.BootstrapTableTool.setupOfflineData(BootstrapTableTool.java:253) ~[classes/:?]
	at org.apache.pinot.tools.BootstrapTableTool.bootstrapOfflineTable(BootstrapTableTool.java:194) ~[classes/:?]
	at org.apache.pinot.tools.BootstrapTableTool.execute(BootstrapTableTool.java:104) ~[classes/:?]
	at org.apache.pinot.tools.admin.command.QuickstartRunner.bootstrapTable(QuickstartRunner.java:232) ~[classes/:?]
	at org.apache.pinot.tools.Quickstart.execute(Quickstart.java:86) ~[classes/:?]
	at org.apache.pinot.tools.GenericQuickstart.execute(GenericQuickstart.java:78) ~[classes/:?]
	at org.apache.pinot.tools.GenericQuickstart.main(GenericQuickstart.java:83) ~[classes/:?]
Caused by: java.lang.RuntimeException: Failed to generate Pinot segment for file - file:/Users/yash/Repos/pinot/pinot-tools/target/classes/examples/batch/fineFoodReviews/rawdata/fine_food_reviews_with_embeddings_1k.parquet.gzip
	at org.apache.pinot.plugin.ingestion.batch.standalone.SegmentGenerationJobRunner.lambda$submitSegmentGenTask$1(SegmentGenerationJobRunner.java:287) ~[classes/:?]
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539) ~[?:?]
	at java.base/java.util.concurrent.FutureTask.run$$$capture(FutureTask.java:264) ~[?:?]
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java) ~[?:?]
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?]
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:?]
	at java.base/java.lang.Thread.run(Thread.java:840) [?:?]
Caused by: java.lang.NoClassDefFoundError: org/apache/hadoop/shaded/com/ctc/wstx/io/InputBootstrapper
	at org.apache.pinot.plugin.inputformat.parquet.ParquetUtils.getParquetHadoopConfiguration(ParquetUtils.java:91) ~[classes/:?]
	at org.apache.pinot.plugin.inputformat.parquet.ParquetNativeRecordReader.init(ParquetNativeRecordReader.java:66) ~[classes/:?]
	at org.apache.pinot.spi.data.readers.RecordReaderFactory.getRecordReaderByClass(RecordReaderFactory.java:148) ~[classes/:?]
	at org.apache.pinot.segment.local.segment.creator.impl.SegmentIndexCreationDriverImpl.getRecordReader(SegmentIndexCreationDriverImpl.java:144) ~[classes/:?]
	at org.apache.pinot.segment.local.segment.creator.impl.SegmentIndexCreationDriverImpl.init(SegmentIndexCreationDriverImpl.java:120) ~[classes/:?]
	at org.apache.pinot.plugin.ingestion.batch.common.SegmentGenerationTaskRunner.run(SegmentGenerationTaskRunner.java:115) ~[classes/:?]
	at org.apache.pinot.plugin.ingestion.batch.standalone.SegmentGenerationJobRunner.lambda$submitSegmentGenTask$1(SegmentGenerationJobRunner.java:265) ~[classes/:?]
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539) ~[?:?]
	at java.base/java.util.concurrent.FutureTask.run$$$capture(FutureTask.java:264) ~[?:?]
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java) ~[?:?]
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?]
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:?]
	at java.base/java.lang.Thread.run(Thread.java:840) ~[?:?]
Caused by: java.lang.ClassNotFoundException: org.apache.hadoop.shaded.com.ctc.wstx.io.InputBootstrapper
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641) ~[?:?]
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188) ~[?:?]
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:525) ~[?:?]
	at org.apache.pinot.plugin.inputformat.parquet.ParquetUtils.getParquetHadoopConfiguration(ParquetUtils.java:91) ~[classes/:?]
	at org.apache.pinot.plugin.inputformat.parquet.ParquetNativeRecordReader.init(ParquetNativeRecordReader.java:66) ~[classes/:?]
	at org.apache.pinot.spi.data.readers.RecordReaderFactory.getRecordReaderByClass(RecordReaderFactory.java:148) ~[classes/:?]
	at org.apache.pinot.segment.local.segment.creator.impl.SegmentIndexCreationDriverImpl.getRecordReader(SegmentIndexCreationDriverImpl.java:144) ~[classes/:?]
	at org.apache.pinot.segment.local.segment.creator.impl.SegmentIndexCreationDriverImpl.init(SegmentIndexCreationDriverImpl.java:120) ~[classes/:?]
	at org.apache.pinot.plugin.ingestion.batch.common.SegmentGenerationTaskRunner.run(SegmentGenerationTaskRunner.java:115) ~[classes/:?]
	at org.apache.pinot.plugin.ingestion.batch.standalone.SegmentGenerationJobRunner.lambda$submitSegmentGenTask$1(SegmentGenerationJobRunner.java:265) ~[classes/:?]
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539) ~[?:?]
	at java.base/java.util.concurrent.FutureTask.run$$$capture(FutureTask.java:264) ~[?:?]
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java) ~[?:?]
```
- The shaded woodstox dependency comes from `hadoop-client-runtime` which wasn't in the dependency tree for `pinot-parquet` before this change. There's not many additional transitive dependencies being brought in through this -
```
 +- org.apache.hadoop:hadoop-client-runtime:jar:3.3.4:compile
 |  +- org.apache.hadoop:hadoop-client-api:jar:3.3.4:runtime
 |  \- commons-logging:commons-logging:jar:1.1.3:runtime
```

### CODE REVIEW

**Review by gortiz — APPROVED** [review]

**Review by gortiz — CHANGES_REQUESTED** [review]
We should not use exclusions. Instead we should indicate dependencies as dependency management. In this case, we should include `hadoop-client-api` and probably `hadoop-client-runtime` in the dependency management section.

By doing so, Maven will use that version instead the one that is brought by `orc-core`

**Comment by yashmayya** [comment]
Thanks @gortiz. `hadoop-client-runtime` was already added to the parent POM's dependency management section. I agree that adding `hadoop-client-api` explicitly as well and avoiding individual exclusions is a better solution overall - I've made that change.

File: pom.xml [hunk 1]
```diff
@@ -1069,6 +1069,24 @@
...
+            <groupId>commons-logging</groupId>
+            <artifactId>commons-logging</artifactId>
+          </exclusion>
+        </exclusions>
+      </dependency>
+      <dependency>
+        <groupId>org.apache.hadoop</groupId>
+        <artifactId>hadoop-client-api</artifactId>
+        <version>${hadoop.version}</version>
+        <scope>provided</scope>
```
Comment (gortiz) [inline pom.xml:1088]: Are we sure the scope needs to be provided? Someone has to actually bring that dependency if it is actually used.

File: pom.xml [hunk 2]
```diff
@@ -1069,6 +1069,24 @@
...
+        <groupId>org.apache.hadoop</groupId>
+        <artifactId>hadoop-client-runtime</artifactId>
+        <version>${hadoop.version}</version>
+        <scope>provided</scope>
+        <exclusions>
+          <exclusion>
+            <groupId>commons-logging</groupId>
+            <artifactId>commons-logging</artifactId>
+          </exclusion>
+        </exclusions>
```
Comment (gortiz) [inline pom.xml:1082]: We should try to do not add exclusions. Are we sure we need to exclude here? Again, dependency management should fix most of the issues we resolved in the past by using exclusions

Comment (yashmayya) [reply pom.xml:1082]: This is an explicitly banned dependency here - https://github.com/apache/pinot/blob/4b5bd217e4a70bc76f60d73897a0484f19955ed5/pom.xml#L1742-L1754

Without this exclusion, the enforcer plugin's enforce goal fails in the build. We're excluding the `commons-logging` dependency from various other dependencies due to this, for example - https://github.com/apache/pinot/blob/4b5bd217e4a70bc76f60d73897a0484f19955ed5/pom.xml#L688-L698

**Review by gortiz — APPROVED** [review]

**Pull request merged**